lib/applicationModel.js

```diff
@@ -89,54 +90,19 @@ class ApplicationModel {
   }  
 
   setupModelClassMap() {
-    if (!this.modelClassMap) {
-      this.modelClassMap = new Map();
-      this.nameToSchemaMap = new Map();
-      // Register all schemas first, then check the anonymous schemas for duplicates
-      ApplicationModel.asyncapi.allSchemas().forEach((schema, name) => {
-        debugApplicationModel(`setupModelClassMap ${name} type ${schema.type()}`);
-        this.registerSchemaNameToModelClass(schema, name);
-        this.nameToSchemaMap[name] = schema;
-      });
-
-      ApplicationModel.asyncapi.allSchemas().forEach((schema, schemaName) => {
-        debugApplicationModel(`setupModelClassMap anonymous schemas ${schemaName} type ${schema.type()}`);
-        this.registerSchemasInProperties(schema);
-        this.registerSchemasInAllOf(schema);
-      });
-      debugApplicationModel('modelClassMap:');
-      debugApplicationModel(this.modelClassMap);
-    }
-  }
-
-  registerSchemasInProperties(schema) {
-    if (!!Object.keys(schema.properties()).length) {
-      // Each property name is the name of a schema. It should also have an x-parser-schema-id name. We'll be adding duplicate mappings (two mappings to the same model class) since the anon schemas do have names
-      Object.keys(schema.properties()).forEach(property => {
-        const innerSchema = schema.properties()[property];
-        const innerSchemaParserId = innerSchema.ext('x-parser-schema-id');
-        const existingModelClass = this.modelClassMap[innerSchemaParserId];
-        if (existingModelClass) {
-          this.modelClassMap[property] = existingModelClass;
-        } else {
-          this.registerSchemaNameToModelClass(innerSchema, property);
-        }
-      });
-    }
-  }
-
-  registerSchemasInAllOf(schema) {
-    const allOf = schema.allOf();
-    debugApplicationModel('allOf:');
-    debugApplicationModel(allOf);
-    if (allOf) {
-      allOf.forEach(innerSchema => {
-        const name = innerSchema.ext('x-parser-schema-id');
-        if (this.isAnonymousSchema(name) && innerSchema.type() === 'object') {
-          this.registerSchemaNameToModelClass(innerSchema, name);
-        }
-      });
+    if (this.modelClassMap) {
+      return;
```

Comment:
Same idea with the conditional changed from the code being nested within, to it not being nested.